src/main/java/org/opensearch/security/authtoken/jwt/JwtVendor.java

```diff
@@ -103,7 +94,7 @@ static Tuple<JWK, JWSSigner> createJwkFromSettings(final Settings settings) {
         }
     }
 
-    public ExpiringBearerAuthToken createJwt(
```

Comment:
Ideally, I'd like to keep this generic `public ExpiringBearerAuthToken createJwt(JwtClaimsSet claims);` and rely on the caller to use the correct instance of a JWTVender.

With the changes in this PR, its designed in a way that the same JWTVendor can issue OBO Tokens and API Tokens.

Should we raise a PR against main to make this more generic for the existing OBO use-case?

Reply:
This suggestion makes sense. Let's not split it up/PR against main. Let's just keep changes in this feature branch: it will get picked up with the merge to main later.